File: is_palindrome/is_palindrome.ts
function isPalindrome(str: string): boolean {
  let firstIdx = 0;
  let lastIdx = str.length - 1;
  while (firstIdx <= lastIdx) {
    if (str[firstIdx] !== str[lastIdx]) {
      return false;
    }
    firstIdx++;
    lastIdx--;
  }
  return true;
}

console.log(isPalindrome("aba"));

File: is_palindrome/is_palindrome.go
package main

import (
  "fmt"
)

func isPalindrome(s string) bool {
  firstIdx := 0
  lastIdx := len(s) - 1
  for firstIdx <= lastIdx {
    if s[firstIdx] != s[lastIdx] {
      return false
    }
    firstIdx++
    lastIdx--
  }
  return true
}

func main() {
  fmt.Println(isPalindrome("aba"))
}
